### PR TITLE
background image scaling good without jumping

### LINKED ISSUE #466: Fix aspect-ratio of custom background images
custom background images are always shown as "scale-to-screen" which does not respect the aspect ration.

i would suggest to show background images always with the correct aspect ration and cut parts top/bottom or left/right (so that the image is centered in the screen rectangle).

https://github.com/deltachat/deltachat-android/issues/460#issuecomment-439395579 shows some images

there is some code in the old telegram-ui that handles this by on-the-fly-creating a temporary bitmap cropped from the orignal one iirc.

### PR DESCRIPTION
**Pull request by angelo-fuchs**
fixes #466 

When finished needs to:
1. Keep aspect ratio of background image in landscape and portrait orientation in the chat view
2. Fill screen with image (no empty borders)
3. Maintain position when the keyboard is opened.

### CODE REVIEW

**Comment by angelo-fuchs** [comment]
I don't particularly like this solution as it is very forced.
I'm uncertain if it works as expected on large landscape screens (e.g. Pads) as I don't have any to test it on.

**Comment by angelo-fuchs** [comment]
This introduced a bug in the Background picker.

**Comment by r10s** [comment]
wondering, how whatsapp does that - but that is closed source. signal does not have a background-image, telegram does weird things all the time,  so wondering if there is some source we can peek at :) conversations?

**Comment by angelo-fuchs** [comment]
@r10s I didn't find any workable code (I didn't search too hard, but spend about an hour searching for examples).
This solution at least works and doesn't do anything surprising.
Just take the image, rescale/crop it once per size and display it when that size is needed.

It could be better, but I don't see how without pouring even more effort into it.

File: src/org/thoughtcrime/securesms/components/ScaleStableImageView.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        if (orientation == ORIENTATION_PORTRAIT) {
+            portrait = true;
+        } else if (orientation == ORIENTATION_LANDSCAPE) {
+            portrait = false;
+        } else {
+            Log.i(TAG, "orientation was: " + orientation);
+            return; // something fishy happened.
+        }
+
+        measureViewSize(width, height, oldWidth, oldHeight, portrait);
```
Comment (cyBerta) [inline src/org/thoughtcrime/securesms/components/ScaleStableImageView.java:89]: This line can be moved below the early return block (`if (!(defaultDrawable instanceof BitmapDrawable)) {`...)

Comment (angelo-fuchs) [reply src/org/thoughtcrime/securesms/components/ScaleStableImageView.java]: Agreed, I changed the code.

**Comment by angelo-fuchs** [comment]
I rebased the changes on master and tested again. Works good.

**Comment by r10s** [comment]
i think, sth. went wrong with rebasing, for me, it shows >100 commits now and >1300 changed lines

**Comment by angelo-fuchs** [comment]
Seems like the rebase went wrong, I will manually rebase from master again and open a new PR for it.

**Pull request closed without merging**